python/cudf/cudf/core/column/lists.py

```diff
@@ -198,6 +198,11 @@ def _with_type_metadata(
 
         return self
 
+    def copy(self, deep: bool = True):
+        # Since list columns are immutable, both deep and shallow copies share
+        # the underlying device data and mask.
+        return super().copy(deep=False)
```

Comment:
Is there a potential footgun here in that if a mask column is taken from a non-list column in the construction, then this is a lie.

Reply:
Umm, yes.. but that would be a bug in our construction no? We shouldn't be using a non-list column mask directly to create a list column.

Reply:
I guess I mean if somehow someone writes:
```
build_column(list_data, ...., mask=mask_from_a_non_list_column)
```
And then (much later), we copy the resulting list column, we've violated the contract on the mask column, I think.

Is there a way that this could, plausibly, happen?

Reply:
Or are masks always copied?

Reply:
Usually, masks are to be copied while being constructed from another type of column mask.